src/posix/platform/radio_spinel.cpp

```diff
@@ -1367,6 +1367,12 @@ otError RadioSpinel::Receive(uint8_t aChannel)
         VerifyOrExit(error == OT_ERROR_NONE);
     }
 
+    if (mTxRadioTid != 0)
```

Comment:
How about only allowing this behavior when the radio NCP doesn't support ACK timeout?

Reply:
What would be the advantage? 
Is there a reason to keep the `mTxRadioTid` (in particular since we do set `mTxState = kIdle;` in `RadioSpinel::Receive()`).  I think it'd better to clear all related tx variables.

The transition out from "Transmit" state using a `Receive()` is in general a valid state transition (please see [valid radio state transitions (radio.h)](https://github.com/openthread/openthread/blob/master/include/openthread/platform/radio.h#L213)) even though we don't have a use for it other than ack timeout.

Reply:
I was afraid this would cause wrong ACK, but actually it will not. Yes, thank you for the explanation.